backend/dataall/modules/s3_datasets_shares/services/s3_share_service.py

```diff
@@ -259,11 +261,17 @@ def get_s3_consumption_data(uri):
             }
 
     @staticmethod
-    def list_shared_databases_tables_with_env_group(environmentUri: str, groupUri: str):
+    @ResourcePolicyService.has_resource_permission(LIST_ENVIRONMENT_DATASETS)
+    def list_shared_databases_tables_with_env_group(uri: str, group_uri: str):
         context = get_context()
+        if group_uri not in context.groups:
```

Comment:
This condition (or the similar ones) is widely used.
May be we can make a decorator out of it?

Reply:
E.g. 
![image](https://github.com/user-attachments/assets/9d089639-b719-49cb-9b89-776e69ee9cf4)